src/shaders/cs_bicubic.comp

```diff
@@ -0,0 +1,248 @@
+// References
+// https://www.codeproject.com/Articles/236394/Bi-Cubic-and-Bi-Linear-Interpolation-with-GLSL
+// https://stackoverflow.com/questions/13501081/efficient-bicubic-filtering-code-in-glsl
+
+#version 460
+
+#extension GL_GOOGLE_include_directive : require
+#extension GL_EXT_shader_explicit_arithmetic_types_float16 : require
+
+#include "descriptor_set.h"
+
+layout(
+  local_size_x = 64,
+  local_size_y = 1,
+  local_size_z = 1) in;
+
+layout(push_constant)
+uniform layers_t {
+    uvec4 c0, c1, c2;
+};
+
+#define A_GPU 1
+#define A_GLSL 1
+#define A_HALF 1
+#include "ffx_a.h"
+#include "bicubic.h"
+
+vec4 catMullRom( const float B, const float C, const float x )
+{
+    vec4 n = vec4(1.0, 2.0, 3.0, 4.0) - x;
+    float f = x;
+    if( f < 0.0 )
+    {
+        f = -f;
+    }
+    if( f < 1.0 )
+    {
+        return n * ( ( 12 - 9 * B - 6 * C ) * ( f * f * f ) +
+            ( -18 + 12 * B + 6 *C ) * ( f * f ) +
+            ( 6 - 2 * B ) ) / 6.0;
+    }
+    else if( f >= 1.0 && f < 2.0 )
```

Comment:
I'm not too sure about it but, this branch seems unreachable?
You pass in f (x) as a fract of a pixel coordinates, thus it will never be beyond 0-1?

The original implementation, which I reimplemented here
https://github.com/dolphin-emu/dolphin/pull/11999/files#diff-368d3b7ce4f1890976b9f3b3d7f59df06cc906721d2dd882552a8503b3a40d7bR240
actually passed in fract*m (or n) here, effectively using all branches as it spanned from -1 to 2.
Thought to be honest I haven't 100% understood what the algorithm does so maybe my implementation is all wrong.

Reply:
Thanks for spotting this, I've updated the code and now it calculates the weights as described [here](https://gist.github.com/TheRealMJP/c83b8c0f46b63f3a88a5986f4fa982b1).